samples/sample_vpp/src/sample_vpp_config.cpp

```diff
@@ -77,122 +67,94 @@ mfxStatus ConfigVideoEnhancementFilters( sInputParams* pParams, sAppResources* p
 
     if (enhFilterCount > 0)
     {
-        pResources->extDoUse.NumAlg = enhFilterCount;
-        pResources->extDoUse.AlgList = pResources->tabDoUseAlg;
-        pVppParam->ExtParam[pVppParam->NumExtParam++] = (mfxExtBuffer*)&(pResources->extDoUse);
+        auto doUse = pVppParam->AddExtBuffer<mfxExtVPPDoUse>();
+        doUse->NumAlg = enhFilterCount;
+        doUse->AlgList = pResources->tabDoUseAlg;
```

Comment:
why we need `tabDoUseAlg` with the smart adapter?

Reply:
I try to safe logic, which we used before (you can check comment [1])

Reply:
@dvrogozh , please see lines 32-65. AlgList  is a pointer, we need to initialize it with some memory. For mfxExtBuffer with underlying memory, we allocate this memory outside of extBufferHolder<T> class.